src/StdChains.sol

```diff
@@ -20,71 +22,126 @@ abstract contract StdChains {
         string rpcUrl;
     }
 
-    bool private initialized;
+    // chain alias -> chain data
+    mapping(string => Chain) internal _chains;
+    mapping(string => string) internal _defaultRpcUrls;
+    mapping(uint256 => string) internal _idToAlias;
 
-    // Maps from a chain's key (matching the alias in the `foundry.toml` file) to chain data.
-    mapping(string => Chain) private chains;
-    // Maps from a chain ID to that chain's key name
-    mapping(uint256 => string) private idToKey;
+    // The RPC URL will be fetched from config or defaultRpcUrls if possible.
+    function getChain(string memory chainAlias) internal virtual returns (Chain memory chain) {
+        require(bytes(chainAlias).length != 0, "StdChains getChain(string): Alias cannot be the empty string.");
 
-    function getChain(string memory key) internal virtual returns (Chain memory) {
         initialize();
-        return chains[key];
+        chain = _chains[chainAlias];
+        require(
+            chain.chainId != 0,
+            string(abi.encodePacked("StdChains getChain(string): Chain with alias \"", chainAlias, "\" not found."))
+        );
+
+        withRpcUrl(chainAlias, chain);
     }
 
-    function getChain(uint256 chainId) internal virtual returns (Chain memory) {
+    function getChain(uint256 chainId) internal virtual returns (Chain memory chain) {
+        require(chainId != 0, "StdChains getChain(uint256): Chain ID cannot be 0.");
         initialize();
-        return chains[idToKey[chainId]];
+        string memory chainAlias = _idToAlias[chainId];
+
+        chain = _chains[chainAlias];
+
+        require(
+            chain.chainId != 0,
+            string(abi.encodePacked("StdChains getChain(uint256): Chain with ID ", vm.toString(chainId), " not found."))
+        );
+
+        withRpcUrl(chainAlias, chain);
+    }
+
+    // lookup rpcUrl, in descending order of priority:
+    // current -> config (foundry.toml) -> default
+    function withRpcUrl(string memory chainAlias, Chain memory chain) private view {
```

Comment:
`withRpcUrl` feels like a bad name. maybe `findRpcUrl` or `getRpcUrl`, then return the RPC URL and set it in the parent method